src/conv/solver_finders.cpp

```diff
@@ -257,8 +257,18 @@ static void EvaluateInvokers(Handle& handle,
         const auto invoker = handle.PrepareInvoker(*sol.invoker_factory, sol.construction_params);
         try
         {
-            invoker(handle, invoke_ctx);
-            const auto elapsed = handle.GetKernelTime();
+            invoker(handle, invoke_ctx); // Dry-run once to warm-up.
+
+            constexpr int N_RUNS = 5;
+            using elapsed_t      = decltype(handle.GetKernelTime());
+            auto elapsed         = static_cast<elapsed_t>(0);
+            for(int i = 0; i < N_RUNS; ++i)
+            {
+                invoker(handle, invoke_ctx);
+                elapsed += handle.GetKernelTime();
+            }
+            elapsed /= static_cast<elapsed_t>(N_RUNS);
+
```

Comment:
This is the core of this change. Other stuff is just removals of unused code.

Reply:
@averinevg Maybe, but this depends on too many factors and must be proven before use. But the idea is good.

Reply:
> For cases where the execution time is relatively long (hundreds of milliseconds), the influence of external factors is negligible. Maybe in such cases one launch is enough?

Resolved in #3103